packages/api/endpoints/executions.js

```diff
@@ -79,6 +86,38 @@ async function del(req, res) {
   return res.send({ message: 'Record deleted' });
 }
 
+/**
+ * Get execution history for a single granule or multiple granules
+ *
+ * @param {Object} req - express request object
+ * @param {Object} res - express response object
+ * @returns {Promise<Object>} the promise of express response object
+ */
+async function history(req, res) {
+  const { granuleId } = req.params;
+  const payload = req.body;
+  const granuleIds = granuleId ? [granuleId] : await getGranuleIdsForPayload(payload);
+
+  const knex = await getKnexClient({ env: process.env });
+
+  const executionPgModel = new ExecutionPgModel();
+  const granulePgModel = new GranulePgModel();
+  const granulesExecutionsPgModel = new GranulesExecutionsPgModel();
+
+  const granuleCumulusIds = await granulePgModel
+    .getRecordsCumulusIds(knex, (builder) => builder.whereIn('granule_id', granuleIds));
+
+  const executionCumulusIds = await granulesExecutionsPgModel
+    .getExecutionCumulusIdsFromGranuleCumulusIds(knex, granuleCumulusIds);
```

Comment:
Doing these two things seems like something common enough we might need to do often, so I'd put a helper in `db/lib/granule.ts` to handle just returning execution cumulus IDs for any set of parameters from granules

Reply:
Agreed, done!